File: python3/solutions/bing.py
# Problem: https://open.kattis.com/problems/bing

import sys
# Cmd + D to print the results in the console

lines = sys.stdin.readlines()
lines.pop(0)

dict = {}

for line in lines:
  line = line.rstrip()

  print(0 if line not in dict else dict[line])

  substr = ''
  for char in line:
    substr += char
    dict[substr] = 1 if substr not in dict else dict[substr] + 1

File: javascript/solutions/bing.js
// Problem: https://open.kattis.com/problems/bing

readline();
const dict = {};
while (line = readline()) {
  line = line.trim();
  print(!dict[line] ? '0' : dict[line]);
  let lineLength = line.length;
  let substr = '';
  for (let i = 0; i < line.length; i += 1) {
    let crtChar = line[i];
    substr += crtChar;
    if (!dict[substr]) {
      dict[substr] = 1;
    } else {
      dict[substr] += 1;
    }
  }
}
